src/vs/workbench/contrib/terminal/browser/terminalTabbedView.ts

```diff
@@ -119,6 +120,20 @@ export class TerminalTabbedView extends Disposable {
 				}
 			}
 		}));
+		this._register(this._terminalService.onDidChangeActiveInstance(() => {
+			const terminalScopedContextKeyService = this._terminalService.activeInstance?.scopedContextKeyService;
```

Comment:
This event fires with the active instance, so you should be able to avoid the `?` in `this._terminalService.activeInstance?`